ibm/resource_ibm_database.go

```diff
@@ -0,0 +1,1086 @@
+		}
+		return fmt.Errorf("Error retrieving resource instance: %s", err)
+	}
+	if strings.Contains(instance.State, "removed") {
+		log.Printf("[WARN] Removing instance from TF state because it's now in removed state")
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("tags", instance.Tags)
+	d.Set("name", instance.Name)
+	d.Set("status", instance.State)
+	d.Set("resource_group_id", instance.ResourceGroupID)
+	d.Set("parameters", flatmap.Flatten(instance.Parameters))
+	d.Set("location", instance.RegionID)
+
+	rsCatClient, err := meta.(ClientSession).ResourceCatalogAPI()
+	if err != nil {
+		return err
+	}
+	rsCatRepo := rsCatClient.ResourceCatalog()
+
+	serviceOff, err := rsCatRepo.GetServiceName(instance.ServiceID)
+	if err != nil {
+		return fmt.Errorf("Error retrieving service offering: %s", err)
+	}
+
+	d.Set("service", serviceOff)
+
+	servicePlan, err := rsCatRepo.GetServicePlanName(instance.ServicePlanID)
+	if err != nil {
+		return fmt.Errorf("Error retrieving plan: %s", err)
+	}
+	d.Set("plan", servicePlan)
+
+	icdClient, err := meta.(ClientSession).ICDAPI()
+	if err != nil {
+		return fmt.Errorf("Error getting database client settings: %s", err)
+	}
+
+	icdId := EscapeUrlParm(instanceID)
+	cdb, err := icdClient.Cdbs().GetCdb(icdId)
+	if err != nil {
+		if apiErr, ok := err.(bmxerror.RequestFailure); ok && apiErr.StatusCode() == 404 {
+			return fmt.Errorf("The database instance was not found in the region set for the Provider. Specify the correct region in the provider definition. %v", err)
+		}
+		return fmt.Errorf("Error getting database config for: %s with error %s\n", icdId, err)
+	}
+	d.Set("adminuser", cdb.AdminUser)
+	d.Set("version", cdb.Version)
+
+	groupList, err := icdClient.Groups().GetGroups(icdId)
+	if err != nil {
+		return fmt.Errorf("Error getting database groups: %s", err)
+	}
+	d.Set("groups", flattenIcdGroups(groupList))
+	d.Set("members_memory_allocation_mb", groupList.Groups[0].Memory.AllocationMb)
+	d.Set("members_disk_allocation_mb", groupList.Groups[0].Disk.AllocationMb)
+
+	whitelist, err := icdClient.Whitelists().GetWhitelist(icdId)
+	if err != nil {
+		return fmt.Errorf("Error getting database whitelist: %s", err)
+	}
+	d.Set("whitelist", flattenWhitelist(whitelist))
+
+	var connectionStrings []CsEntry
+	//ICD does not implement a GetUsers API. Users populated from tf configuration.
+	tfusers := d.Get("users").(*schema.Set)
+	users := expandUsers(tfusers)
+	user := icdv4.User{
+		UserName: cdb.AdminUser,
+	}
+	users = append(users, user)
+	for _, user := range users {
+		userName := user.UserName
+		csEntry, err := getConnectionString(d, userName, meta)
+		if err != nil {
+			return fmt.Errorf("Error getting user connection string for user (%s): %s", userName, err)
+		}
+		connectionStrings = append(connectionStrings, csEntry)
+	}
+	d.Set("connectionstrings", flattenConnectionStrings(connectionStrings))
+
+	return nil
+}
+
+func resourceIBMDatabaseInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
"name & "plan" cannot be updated for a db resource instance?

Reply:
Somehow I lost the name update. I've added that back in. Plan cannot be changed as today there is only a single plan "standard". I left plan in the definition to keep it consistent with all other services, but its not really needed.